rosbag2_compression/src/rosbag2_compression/zstd_decompressor.cpp

```diff
@@ -156,6 +158,52 @@ void write_output_buffer(
   fclose(file_pointer);
 }
 
+/**
+ * Throws a runtime_error if data does not contain a ZSTD compressed frame.
+ */
+void throw_on_invalid_magic_number(const std::vector<uint8_t> & data)
+{
+  if (data.size() < 4) {
```

Comment:
Why 4? Another magic number? 😅

Reply:
would it be cleaner if it was `sizeof(uint32_t)` ?